packages/blocks/callout/src/App.tsx

```diff
@@ -0,0 +1,59 @@
+import { BlockComponent } from "blockprotocol/react";
+import React, { CSSProperties, RefCallback, useCallback } from "react";
+
+import { EmojiIcon } from "./EmojiIcon";
+
+type AppProps = {
+  editableRef?: RefCallback<HTMLElement>;
+  icon?: string;
+  text?: string;
+};
+
+const wrapperStyle: CSSProperties = {
+  borderRadius: "0.25em",
+  width: "100%",
+  position: "relative",
+  padding: "0.5em",
+  background: "#f9fafc",
+  border: "1px solid #dee7f3",
+};
+
+const textStyle: CSSProperties = {
+  minHeight: "1.5em",
+  paddingLeft: "1.5em",
+};
+
+export const App: BlockComponent<AppProps> = ({
+  editableRef,
+  icon,
+  text,
+  entityId,
+  updateEntities,
+}) => {
+  const handleIconChange = useCallback(
+    (newIcon: string | undefined): void => {
+      if (!entityId) {
+        return;
+      }
+
+      void updateEntities?.([
+        {
+          entityId,
+          data: { icon: newIcon ?? null },
+        },
+      ]);
+    },
+    [entityId, updateEntities],
+  );
+
+  return (
+    <div style={wrapperStyle}>
+      <EmojiIcon value={icon} onChange={handleIconChange} />
+      {editableRef ? (
+        <div style={textStyle} ref={editableRef} />
+      ) : (
+        <div style={textStyle}>{text}</div>
+      )}
```

Comment:
Oh yeah you are right! I removed the ternary operator. I thought we'd have more distinct nodes when I added it and forgot to reassess later.

Reply:
UPD: I had to make `text` a conditional child:

```
      <div style={textStyle} ref={editableRef}>
        {editableRef ? undefined : text}
      </div>
```

Was getting this otherwise: 

<img width="690" alt="Screenshot 2022-02-16 at 18 00 35" src="https://user-images.githubusercontent.com/608862/154327392-c0a7d3d8-a332-4e06-bd61-684317ed622b.png">
<img width="467" alt="Screenshot 2022-02-16 at 18 00 39" src="https://user-images.githubusercontent.com/608862/154327395-c33197a9-813e-4bff-beae-bd1ef80a210b.png">

Reply:
I also kept `textStyle` outside the component. There is no point redefining it inline on each render. Having the same object ref should skip object to string conversion on each render. The difference is tiny, but why not?